test/partialAttribute/nested-html/stamped_from_polymer_template.html

```diff
@@ -157,147 +157,140 @@
+                        importedTemplate.addEventListener('stamping', function onceStamped(){
+                            expect(importedTemplate.model.scope1).to.equal(changedScope);
+                            done();
+                        });
                     });
                 });
-            });
 
-            describe('after `dom-bind` re-set the sub-partial property (just one scope) to the same instance', function () {
-                var scope;
+                describe('after `dom-bind` re-set the sub-partial (just one scope) to the same instance', function () {
+                    var scope;
 
-                beforeEach(function(done) {
-                    // IDEA: promisify juicy-html
-                    sinon.stub(importedTemplate, "_setPendingPropertyOrPath");
-                    importedTemplate.addEventListener('stamping', function onceStamped(){
-                        importedTemplate.removeEventListener('stamping', onceStamped);
+                    beforeEach(function() {
+                        sinon.stub(importedTemplate, "_setPendingPropertyOrPath");
                         scope = domBind.get('model.Page.scope1');
                         domBind.set('model.Page.scope1', scope);
-                        done();
                     });
-                });
 
-                it('should NOT forward the notification to `imported-template` - Polymer limitation', function () {
-                    expect(importedTemplate._setPendingPropertyOrPath).not.to.be.called;
+                    it('should NOT forward the notification to `imported-template` - Polymer limitation', function () {
+                        expect(importedTemplate._setPendingPropertyOrPath).not.to.be.called;
+                    });
                 });
-            });
 
-            describe('after `dom-bind` changes the sub-partial property of a view with an explicit composition', function () {
-                var scope, include, newVal = "yeah", customComposition = "my custom composition";
+                describe('after `dom-bind` changes the sub-partial\'s property of a view with an explicit composition', function () {
+                    var scope, include, newVal = "yeah", customComposition = "my custom composition", shadowRoot;
 
-                beforeEach(function(done) {
-                    sinon.stub(importedTemplate, "_setPendingPropertyOrPath");
-                    importedTemplate.addEventListener('stamping', function onceStamped(){
-                        importedTemplate.removeEventListener('stamping', onceStamped);
+                    beforeEach(function() {
+                        sinon.stub(importedTemplate, "_setPendingPropertyOrPath");
                         include = container.querySelector('starcounter-include');
+                        // TODO: do we really want to tangle starcounter-layout-html-editor in the case that should be atomic as well?
```

Comment:
The test is actually atomic, it only mentions `starcounter-layout-html-editor` in a comment. I agree that the comment `//this is how starcounter-layout-html-editor uses it` can be moved from the test to JSDoc above the `compositionChanged` method, if this is what you mean.

I find it highly confusing that this method looks like "private" even though it is called by `starcounter-layout-html-editor`  and I think it requires explanation.

Reply:
I mean, I'd create such a use case that tests actualyy:

Reply:
>I find it highly confusing that this method looks like "private" even though it is called by starcounter-layout-html-editor and I think it requires explanation.

Reply:
I agree the comment should be rephrased, as it does not affect atomicity, rather testing private methods for external usecase. How about:

Reply:
I created a PR that covers also the refactored code https://github.com/Starcounter/starcounter-include/pull/78, if that's the way to go feel free to close this one.

Reply:
I agree. In the right world, CompositionEditor and ComposiitonProvider communicate through the database,

Reply:
> In the right world, CompositionEditor and ComposiitonProvider should communicate through the database, not through a (private!) method.